ml_inference_offloading/src/main/java/ai/nnstreamer/ml/inference/offloading/MainService.kt

```diff
@@ -83,6 +86,25 @@ class MainService : Service() {
     private lateinit var container: ServiceContainer
     private var initialized = false
     private var serverInfoMap = mutableMapOf<String, ServerInfo>()
+    private lateinit var nsdManager: NsdManager
+
+    private val registrationListener = object : NsdManager.RegistrationListener {
```

Comment:
It looks OK, but is it necessary to create a sub-class of the object and use it as an argument?

Reply:
Not necessary..
Also, it makes `listener already in use` error.

Reply:
@niley7464 Does it need to be Singleton?

Reply:
Do you mean `nsdManager`?

Reply:
@niley7464 The instance of ```NsdManager.RegistrationListener```

Reply:
I think listener should not be Singleton, so it is created everytime when registering service